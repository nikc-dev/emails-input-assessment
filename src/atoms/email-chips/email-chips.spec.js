import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EmailChips from './index';



it('should render EmailChips correctly for Valid email', () => {
  const component = shallow(
    <EmailChips removeEmail={jest.fn()} email={{emailId: 'test@test.com'}} />,
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should render EmailChips correctly for InValid email', () => {
    const component = shallow(
      <EmailChips removeEmail={jest.fn()} email={{emailId: 'test@test.com', isInValid: true}} />,
    );
    expect(toJson(component)).toMatchSnapshot();
});

it('should remove EmailChip correctly onClick of remove button', () => {
    const component = shallow(
      <EmailChips removeEmail={jest.fn()} email={{emailId: 'test@test.com', isInValid: true}} />,
    );
    component.find('#emailChips__remove-btn').simulate('click')
    expect(toJson(component)).toMatchSnapshot();
});
