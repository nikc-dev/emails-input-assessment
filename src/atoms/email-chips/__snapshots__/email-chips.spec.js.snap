// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should remove EmailChip correctly onClick of remove button 1`] = `
<span
  className="emailChips__invalid-email"
>
  test@test.com
  <span
    className="emailChips__remove-btn"
    id="emailChips__remove-btn"
    onClick={[Function]}
  >
    X
  </span>
</span>
`;

exports[`should render EmailChips correctly for InValid email 1`] = `
<span
  className="emailChips__invalid-email"
>
  test@test.com
  <span
    className="emailChips__remove-btn"
    id="emailChips__remove-btn"
    onClick={[Function]}
  >
    X
  </span>
</span>
`;

exports[`should render EmailChips correctly for Valid email 1`] = `
<span
  className="emailChips"
>
  test@test.com
  <span
    className="emailChips__remove-btn"
    id="emailChips__remove-btn"
    onClick={[Function]}
  >
    X
  </span>
</span>
`;
